reflect/src/main/java/dagger/reflect/ReflectiveAssistedFactoryLookupFactory.java

```diff
@@ -0,0 +1,263 @@
+package dagger.reflect;
+
+import static dagger.reflect.Reflection.boxIfNecessary;
+import static dagger.reflect.Reflection.findScope;
+import static dagger.reflect.TypeUtil.canonicalize;
+
+import com.google.auto.value.AutoValue;
+import dagger.assisted.Assisted;
+import dagger.assisted.AssistedFactory;
+import dagger.assisted.AssistedInject;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import javax.inject.Qualifier;
+import org.jetbrains.annotations.Nullable;
+
+final class ReflectiveAssistedFactoryLookupFactory implements JustInTimeLookup.Factory {
+  @Override
+  public @Nullable JustInTimeLookup create(Key key) {
+    if (key.qualifier() != null) {
+      return null; // AssistedFactory type cannot be qualified
+    }
+
+    Type type = key.type();
+
+    try {
+      return getJustInTimeAssistedFactoryLookup(type);
+    } catch (IllegalStateException t) {
+      throw new IllegalStateException(
+          String.format("Failed to create @AssistedFactory %s: \n%s", type, t.getMessage()), t);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.